tests/accessor_tests.cpp

```diff
@@ -0,0 +1,177 @@
+    auto jsonData = std::make_unique<fastgltf::GltfDataBuffer>();
+    REQUIRE(jsonData->loadFromFile(simpleSparseAccessor / "SimpleSparseAccessor.gltf"));
+
+    fastgltf::Parser parser;
+    auto model = parser.loadGLTF(jsonData.get(), simpleSparseAccessor, fastgltf::Options::LoadExternalBuffers);
+    REQUIRE(model != nullptr);
+    REQUIRE(parser.getError() == fastgltf::Error::None);
+    REQUIRE(model->parse(fastgltf::Category::Buffers | fastgltf::Category::BufferViews
+			| fastgltf::Category::Accessors) == fastgltf::Error::None);
+    REQUIRE(model->validate() == fastgltf::Error::None);
+
+    auto asset = model->getParsedAsset();
+    REQUIRE(asset->accessors.size() == 2);
+    REQUIRE(!asset->accessors[0].sparse.has_value());
+    REQUIRE(asset->accessors[1].sparse.has_value());
+    auto& sparse = asset->accessors[1].sparse.value();
+    REQUIRE(sparse.count == 3);
+    REQUIRE(sparse.bufferViewIndices == 2);
+    REQUIRE(sparse.byteOffsetIndices == 0);
+    REQUIRE(sparse.bufferViewValues == 3);
+    REQUIRE(sparse.byteOffsetValues == 0);
+    REQUIRE(sparse.indexComponentType == fastgltf::ComponentType::UnsignedShort);
+
+
+	auto& secondAccessor = asset->accessors[1];
+	auto& viewIndices = asset->bufferViews[secondAccessor.sparse->bufferViewIndices];
+	auto& viewValues = asset->bufferViews[secondAccessor.sparse->bufferViewValues];
+
+	auto& viewData = asset->bufferViews[*secondAccessor.bufferViewIndex];
+	auto* bufferData = getBufferData(asset->buffers[viewData.bufferIndex]) + viewData.byteOffset
+			+ secondAccessor.byteOffset;
+	auto dataStride = viewData.byteStride ? *viewData.byteStride
+			: fastgltf::getElementByteSize(secondAccessor.type, secondAccessor.componentType);
+
+	auto* dataIndices = reinterpret_cast<const std::uint16_t*>(getBufferData(asset->buffers[viewIndices.bufferIndex])
+			+ viewIndices.byteOffset + secondAccessor.sparse->byteOffsetIndices);
+	auto* dataValues = reinterpret_cast<const glm::vec3*>(getBufferData(asset->buffers[viewValues.bufferIndex])
+			+ viewValues.byteOffset + secondAccessor.sparse->byteOffsetValues);
+
+	auto checkValues = std::make_unique<glm::vec3[]>(secondAccessor.count);
+
+	for (std::size_t i = 0, sparseIndex = 0; i < secondAccessor.count; ++i) {
+		if (sparseIndex < secondAccessor.sparse->count && dataIndices[sparseIndex] == i) {
+			checkValues[i] = dataValues[sparseIndex];
+			++sparseIndex;
+		} else {
+			checkValues[i] = *reinterpret_cast<const glm::vec3*>(bufferData + dataStride * i);
+		}
+	}
+
+	SECTION("getAccessorElement") {
+		for (std::size_t i = 0; i < secondAccessor.count; ++i) {
+			REQUIRE(checkValues[i] == fastgltf::getAccessorElement<glm::vec3>(*asset, secondAccessor, i));
+		}
+	}
+
+	SECTION("iterateAccessor") {
+		auto dstCopy = std::make_unique<glm::vec3[]>(secondAccessor.count);
+		std::size_t i = 0;
+
+		fastgltf::iterateAccessor<glm::vec3>(*asset, secondAccessor, [&](auto&& v3) {
+			dstCopy[i++] = std::move(v3);
```

Comment:
Use `std::forward`.

Reply:
Fixed in [df17edc](https://github.com/spnda/fastgltf/pull/20/commits/df17edc742f1893ece80c0cc2467a9b68b7ec880)